src/tribler-core/tribler_core/modules/metadata_store/community/remote_query_community.py

```diff
@@ -171,23 +197,30 @@ async def on_remote_select_response(self, peer, response_payload):
         processing_results = await self.mds.process_compressed_mdblob_threaded(response_payload.raw_blob)
         self.logger.info(f"Response result: {processing_results}")
 
-        # If we now about updated versions of the received stuff, push the updates back
+        # If we know about updated versions of the received stuff, push the updates back
         if isinstance(request, SelectRequest) and self.settings.push_updates_back_enabled:
-            newer_entities = [md for md, result in processing_results if result == GOT_NEWER_VERSION]
+            newer_entities = [r.md_obj for r in processing_results if r.obj_state == ObjState.GOT_NEWER_VERSION]
             self.send_db_results(peer, response_payload.id, newer_entities)
 
         # Query back the sender for preview contents for the new channels
         # TODO: maybe transform this into a processing_callback?
         if self.settings.channel_query_back_enabled:
-            new_channels = [md for md, result in processing_results if result in (UNKNOWN_CHANNEL, UNKNOWN_COLLECTION)]
-            for channel in new_channels:
-                request_dict = {
-                    "channel_pk": hexlify(channel.public_key),
-                    "origin_id": channel.id_,
-                    "first": 0,
-                    "last": self.settings.max_channel_query_back,
-                }
-                self.send_remote_select(peer=peer, **request_dict)
+            for result in processing_results:
+                if result.obj_state == ObjState.UNKNOWN_OBJECT and result.md_obj.metadata_type in (
+                    CHANNEL_TORRENT,
+                    COLLECTION_NODE,
+                ):
+                    request_dict = {
+                        "metadata_type": [COLLECTION_NODE, REGULAR_TORRENT],
+                        "channel_pk": hexlify(result.md_obj.public_key),
+                        "origin_id": result.md_obj.id_,
+                        "first": 0,
+                        "last": self.settings.max_channel_query_back,
+                    }
+                    self.send_remote_select(peer=peer, **request_dict)
+                for dep_query_dict in result.missing_deps:
+                    # Query back for missing dependencies, e.g. thumbnail/description
+                    self.send_remote_select(peer=peer, **dep_query_dict)
```

Comment:
I guess you missed that line 221 `for dep_query_dict ...` must still be processed even in case the `UNKNOWN_OBJECT` condition does not trigger. So, inverting the condition will break the app logic.